### ISSUE TITLE
Modal title missing a heading element (1.3.1)

### ISSUE THREAD

**Issue by Blackbaud-MattGregg** [blackbaud/skyux-modals#93]
### Issue from RE NXT A11y audit in 2019
Avoiding the use of implicit headings -
"New Dashboard" is not an HTML heading element.

```
<div _ngcontent-c20="" class="sky-modal-header-content sky-emphasized sky-section-heading" id="sky-modal-header-id-1">
<sky-modal-header _ngcontent-c23="">
New dashboard        
</sky-modal-header>
</div>
```
### Impact
In the absence of a proper heading structure, users may have difficulty understanding the page structure.

### Expected behavior
Developers must ensure headings are assigned properly using HTML heading elements.
Modal heading (i.e., New dashboard in example) is within a html heading element (e.g., h1)

**Comment by Blackbaud-AlexKingman** [comment]
@Blackbaud-MattGregg Typically the consumer can control the contents of the modal header. Are you suggesting the modal component should enforce a heading element? We could build it so the user's content, in this case "New dashboard", is wrapped in the proper element. If so, is it safe to always use an `h1`, or do we need more context to make that decision for them?

**Comment by Blackbaud-MattGregg** [comment]
@Blackbaud-AlexKingman  Yes, it should include/enforce a heading element. That way teams don't have to think about it. That was the idea anyway.

I did some looking into heading levels and guidance from W3C and other design systems. There are 2 schools of thought in the discussion thread I found on this topic and no clear winner btw using an H1 and H2. Each has rationale....
1.  Modal window is child of page and so it should be H2 b/c page has H1 heading as the parent.
2.  It's a new context and it should start over.

In our modal case, I thought h1 makes the most sense given the headings in a modal content will then follow in the hierarchy and teams can start using H2 and I guess I subscribe to a new heading hierarchy being useful for screen reader users in the new modal context (# 2)

**Comment by Blackbaud-AlexKingman** [comment]
Thanks @Blackbaud-MattGregg. We'll add an `h1` to the modal header. 👍
